test/e2e/e2ev2.go

```diff
@@ -0,0 +1,219 @@
+
+			By("uninstalling cert manager bundle")
+			kbc.UninstallCertManager()
+
+			By("remove container image and work dir")
+			kbc.Destroy()
+		})
+
+		It("should generate a runnable project", func() {
+			var controllerPodName string
+			By("init v1 project")
+			err := kbc.Init(
+				"--project-version", "2",
+				"--domain", kbc.Domain,
+				"--dep=false")
+			Expect(err).Should(Succeed())
+
+			By("creating api definition")
+			err = kbc.CreateAPI(
+				"--group", kbc.Group,
+				"--version", kbc.Version,
+				"--kind", kbc.Kind,
+				"--namespaced",
+				"--resource",
+				"--controller",
+				"--make=false")
+			Expect(err).Should(Succeed())
+
+			By("implementing the API")
+			Expect(insertCode(
+				filepath.Join(kbc.Dir, "api", kbc.Version, fmt.Sprintf("%s_types.go", strings.ToLower(kbc.Kind))),
+				fmt.Sprintf(`type %sSpec struct {
+`, kbc.Kind),
+				`	// +optional
+	Count int `+"`"+`json:"count,omitempty"`+"`"+`
+`)).Should(Succeed())
+
+			By("implementing the mutating and validating webhooks")
+			err = (&scaffold.Webhook{
+				Domain:    kbc.Domain,
+				Group:     kbc.Group,
+				Version:   kbc.Version,
+				Kind:      kbc.Kind,
+				Resources: kbc.Resources,
+			}).WriteTo(filepath.Join(
+				kbc.Dir, "api", kbc.Version,
+				fmt.Sprintf("%s_webhook.go", strings.ToLower(kbc.Kind))))
+			Expect(err).Should(Succeed())
+
+			By("uncomment kustomization.yaml to enable webhook and ca injection")
+			Expect(uncommentCode(
+				filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
+				"#- ../webhook", "#")).To(Succeed())
+			Expect(uncommentCode(
+				filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
+				"#- ../certmanager", "#")).To(Succeed())
+			Expect(uncommentCode(
+				filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
+				"#- manager_webhook_patch.yaml", "#")).To(Succeed())
+			Expect(uncommentCode(
+				filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
+				"#- webhookcainjection_patch.yaml", "#")).To(Succeed())
+
+			By("building image")
+			err = kbc.Make("docker-build", "IMG="+kbc.ImageName)
+			Expect(err).Should(Succeed())
+
+			By("loading docker image into kind cluster")
+			err = kbc.LoadImageToKindCluster()
+			Expect(err).Should(Succeed())
+
+			// NOTE: If you want to run the test against a GKE cluster, you will need to grant yourself permission.
+			// Otherwise, you may see "... is forbidden: attempt to grant extra privileges"
+			// $ kubectl create clusterrolebinding myname-cluster-admin-binding --clusterrole=cluster-admin --user=myname@mycompany.com
+			// https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control
+			By("deploying controller manager")
+			err = kbc.Make("deploy")
+			Expect(err).Should(Succeed())
+
+			By("validate the controller-manager pod running as expected")
+			verifyControllerUp := func() error {
+				// Get pod name
+				podOutput, err := kbc.Kubectl.GetPods(
```

Comment:
wondering why we can't use the `Kubectl.Get` method for this ?

Reply:
We use `GetPods` several times. I thought it may be helpful to have a helper for it.
If it doesn't help much, I will change just use `Get`